pkg/service/utgen/gen.go

```diff
@@ -235,13 +259,29 @@ func (g *UnitTestGenerator) Start(ctx context.Context) error {
 					return fmt.Errorf("process cancelled by user")
 				default:
 				}
-				err = g.ValidateTest(generatedTest, &passedTests, &noCoverageTest, &failedBuild, installedPackages)
+				coverageInc, err := g.ValidateTest(generatedTest, &passedTests, &noCoverageTest, &failedBuild, installedPackages)
 				if err != nil {
 					utils.LogError(g.logger, err, "Error validating test")
+
+					rerr := revertSourceCode(g.srcPath, originalSrcCode, codeModified)
+					if rerr != nil {
+						utils.LogError(g.logger, rerr, "Error reverting source code")
+					}
+
 					return err
 				}
+
+				// if any test increases the coverage, set the flag to true
+				overallCovInc = overallCovInc || coverageInc
 			}
 
+			// if any of the test couldn't increase the coverage, revert the source code
+			if !overallCovInc {
+				err := revertSourceCode(g.srcPath, originalSrcCode, codeModified)
```

Comment:
instead of using this boolean can't we use the error the way we do for test file changes